spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/config/xml/HazelcastEventDrivenInboundChannelAdapterParser.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.config.xml;
+
+import org.w3c.dom.Element;
+
+import reactor.util.Assert;
+import reactor.util.StringUtils;
+
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.xml.AbstractSingleBeanDefinitionParser;
+import org.springframework.beans.factory.xml.ParserContext;
+import org.springframework.integration.config.xml.IntegrationNamespaceUtils;
+import org.springframework.integration.hazelcast.common.CacheEventType;
+import org.springframework.integration.hazelcast.common.HazelcastIntegrationDefinitionValidator;
+import org.springframework.integration.hazelcast.inbound.HazelcastEventDrivenMessageProducer;
+
+/**
+ * Hazelcast Event Driven Inbound Channel Adapter Parser is a BeanDefinitionParser
+ * implementation. It parses int-hazelcast:inbound-channel-adapter xml definition and
+ * defines just a single BeanDefinition.
+ * 
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ *
+ */
+public class HazelcastEventDrivenInboundChannelAdapterParser extends AbstractSingleBeanDefinitionParser {
+
+	private static final String CHANNEL_ATTRIBUTE = "channel";
+
+	private static final String CACHE_ATTRIBUTE = "cache";
+
+	private static final String CACHE_EVENTS_ATTRIBUTE = "cache-events";
+
+	private static final String CACHE_LISTENING_POLICY_ATTRIBUTE = "cache-listening-policy";
+
+	private static final String OUTPUT_CHANNEL = "outputChannel";
+
+	private static final String DISTRIBUTED_OBJECT = "distributedObject";
+
+	private static final String CACHE_EVENT_TYPES = "cacheEventTypes";
+
+	private static final String CACHE_LISTENING_POLICY = "cacheListeningPolicy";
+
+	@Override
+	protected String getBeanClassName(Element element) {
+		return HazelcastEventDrivenMessageProducer.class.getName();
+	}
+
+	@Override
+	protected boolean shouldGenerateIdAsFallback() {
+		return false;
+	}
+
+	@Override
+	protected boolean shouldGenerateId() {
+		return true;
+	}
```

Comment:
Instead of these two wrong methods, please, use `resolveId`. You can simply copy/paste it from the `WebSocketInboundChannelAdapterParser`.
And I guess the same for all other `HazelcastMessageProducer` implementors.